src/haven/motor_position.py

```diff
@@ -99,40 +109,22 @@ def save_motor_position(*motors, name: str, collection=None):
       save.
     name
       A human-readable name for this position (e.g. "sample center")
-    collection
-      A pymongo collection object to receive the data. Meant for
-      testing.
+    md
+      Additional metadata to store with the motor position.
 
-    Returns
-    =======
-    item_id
-      The ID of the item in the database.
     """
-    # Get default collection if none was given
-    if collection is None:
-        collection = default_collection()
     # Resolve device names or labels
     motors = registry.findall(motors)
-    # Prepare the motor positions
-    motor_axes = []
-    for m in motors:
-        payload = dict(name=m.name, readback=rbv(m))
-        # Save the calibration offset for motors
-        if hasattr(m, "user_offset"):
-            payload["offset"] = m.user_offset.get()
-        axis = MotorAxis(**payload)
-        motor_axes.append(axis)
-    savetime = time.time()
-    position = MotorPosition(name=name, motors=motor_axes, savetime=savetime)
-    # Write to the database
-    pos_id = position.save(collection=collection)
-    log.info(f"Saved motor position {name} (uid={pos_id})")
-    return pos_id
+    # Create the new run object
+    _md = {
+        "position_name": name,
+        "plan_name": "save_motor_position",
+    }
+    _md.update(md)
+    yield from bp.count(motors, md=_md)
```

Comment:
The idea is that we want "position_name" and "plan_name" to be the values specified on lines 120/121, unless they are explicitly given in the *md* argument. If I use ``md`` directly, those values don't get included. This way ``_md`` is a temporary dictionary that then gets updated by the ``md`` argument on line 123. If e.g. *plan_name* is a key in ``md``, then it will override the one on line 121, if not then the default on line 121 gets used.

Reply:
oh, I see! I misread line 123.. I thought it was updating md with _md values

Reply:
Gotcha. Yeah I got that pattern from one of bluesky's plan, it took me a few looks to understand what was going on.

If it were the other way around it would cause problems with the mutable argument you pointed out elsewhere. =)